### DISCUSSION TITLE
suggestion to : Why so little power if it is at 100? What is Smax? Why 1000?

### DISCUSSION THREAD

**Discussion by edsonee** [arkypita/LaserGRBL#1600] [General]
Hello friends, first of all, I thank the creator of this great program. This suggestion is based on a frequently asked question from new users. (I belong to a group of lasergrbl whatsapp and it always asks the same thing)
Why so little power if it is at 100? What is Smax? Why 1000? 

![1](https://user-images.githubusercontent.com/83145998/135501266-e646d04c-4bc9-4fbb-9ae0-4fab3952368a.JPG)
![2](https://user-images.githubusercontent.com/83145998/135501273-a89b9d3c-1df9-4eb4-b201-67adc7aded9a.JPG)
![3](https://user-images.githubusercontent.com/83145998/135501284-b7b9e565-bb9f-4529-80c8-5c98c5c8f333.JPG)

I suggest SMAX work on percentage and name change. (see picture 3)
It is a suggestion designed for the comfort of the user
I know that everything is explained and documented on the same page and old users have no problem with this issue. but new users are always confused.
thank you very much.

**Is your feature request related to a problem? Please describe.**

**Describe the solution you'd like**

**Describe alternatives you've considered**

**Additional context**

**Comment by arkypita** [comment]
https://lasergrbl.com/wp-content/uploads/2020/05/Grbl-Configuration-ENG.pdf

**Comment by edsonee** [comment]
I know that everything is explained and documented on the same page and old users have no problem with this issue. but new users are always confused.
I suggest SMAX work on percentage and name change. (see picture 3)
It is a suggestion designed for the comfort of the user

![3](https://user-images.githubusercontent.com/83145998/135590390-0912a5cb-0d69-4e98-bdf2-c4c02fb890a0.JPG)




________________________________
De: arkypita ***@***.***>
Enviado: viernes, 1 de octubre de 2021 04:19
Para: arkypita/LaserGRBL ***@***.***>
Cc: edsonee ***@***.***>; Author ***@***.***>
Asunto: Re: [arkypita/LaserGRBL] suggestion to : Why so little power if it is at 100? What is Smax? Why 1000? (#1591)


https://lasergrbl.com/wp-content/uploads/2020/05/Grbl-Configuration-ENG.pdf

—
You are receiving this because you authored the thread.
Reply to this email directly, view it on GitHub<https://github.com/arkypita/LaserGRBL/issues/1591#issuecomment-931973967>, or unsubscribe<https://github.com/notifications/unsubscribe-auth/AT2LKDS4EVW3XKX42LDBUDDUEVOJJANCNFSM5FC22IFA>.
Triage notifications on the go with GitHub Mobile for iOS<https://apps.apple.com/app/apple-store/id1477376905?ct=notification-email&mt=8&pt=524675> or Android<https://play.google.com/store/apps/details?id=com.github.android&referrer=utm_campaign%3Dnotification-email%26utm_medium%3Demail%26utm_source%3Dgithub>.

**Comment by edsonee** [comment]
![3](https://user-images.githubusercontent.com/83145998/135590613-9ba83a9f-454f-4d3f-826a-600453fe7d44.JPG)
![1](https://user-images.githubusercontent.com/83145998/135590651-debeeadb-0e0e-49fb-833b-22886758750a.JPG)
![2](https://user-images.githubusercontent.com/83145998/135590666-2ca17c85-62d1-417e-abdd-263d2307da4e.JPG)

**Comment by alastaira** [comment]
@edsonee g-code - the language which LaserGRBL (and all GRBL controllers use), was originally designed to control spindles on CNC machines. As such, the "S-Max" value referred to the maximum rotation speed of the spindle. One spindle might rotate at a max of 10,000RPM, while another had a maximum of 20,000RPM, so the $30 value was designed so that users could specify that, and ensure they wouldn't exceed the maximum.

A speed of 10,000, on a machine capable of spinning at 20,000RPM, would be 50%.
A speed of 10,000 on a machine capable of spinning at 10,000RPM, would be 100%.
But they'd be rotating at the same speed..... so it makes no sense to express as a percentage.

The concept of "rotation speed" doesn't really apply to lasers, so people generally choose either 255 or 1000 as their "maximum rotation speed", and then specify some fraction of that. It's not a LaserGRBL issue - it's just how gcode has been developed.

**Comment by edsonee** [comment]
Thanks for answering, I know that gcode in its programming does not work with percentage ($ 30) but for new users it is easier to express the power of the laser in%, a new user will never say "I want to work at 255 of my laser "or" I want to work at 1000 of my laser ". They don't know what "smax" or "$ 30" is. That is why I suggest that the interface works in% and that internally LASERGRBL does the transformation. example (100% = $ 30 255) or (50% = $ 30 255/2).
additionally you give examples of RMP but LASERGRBL is designed to work with laser, and users tend to say "I want to work at 100% of the power of my laser".

Thank you for reading my comment.



![3](https://user-images.githubusercontent.com/83145998/135730273-8b10d7eb-18ac-48ff-8618-3c59eb8647e8.JPG)


SIP...[http://gfx1.hotmail.com/mail/w4/pr04/ltr/emo/winking_smile.gif]!!!

________________________________
De: alastaira ***@***.***>
Enviado: sábado, 2 de octubre de 2021 15:23
Para: arkypita/LaserGRBL ***@***.***>
Cc: edsonee ***@***.***>; Mention ***@***.***>
Asunto: Re: [arkypita/LaserGRBL] suggestion to : Why so little power if it is at 100? What is Smax? Why 1000? (#1591)


@edsonee<https://github.com/edsonee> g-code - the language which LaserGRBL (and all GRBL controllers use), was originally designed to control spindles on CNC machines. As such, the "S-Max" value referred to the maximum rotation speed of the spindle. One spindle might rotate at a max of 10,000RPM, while another had a maximum of 20,000RPM, so the $30 value was designed so that users could specify that, and ensure they wouldn't exceed the maximum.

A speed of 10,000, on a machine capable of spinning at 20,000RPM, would be 50%.
A speed of 10,000 on a machine capable of spinning at 10,000RPM, would be 100%.
But they'd be rotating at the same speed..... so it makes no sense to express as a percentage.

The concept of "rotation speed" doesn't really apply to lasers, so people generally choose either 255 or 1000 as their "maximum rotation speed", and then specify some fraction of that. It's not a LaserGRBL issue - it's just how gcode has been developed.

—
You are receiving this because you were mentioned.
Reply to this email directly, view it on GitHub<https://github.com/arkypita/LaserGRBL/issues/1591#issuecomment-932798358>, or unsubscribe<https://github.com/notifications/unsubscribe-auth/AT2LKDRVWSQJKIZSSOEUDVTUE5E3JANCNFSM5FC22IFA>.
Triage notifications on the go with GitHub Mobile for iOS<https://apps.apple.com/app/apple-store/id1477376905?ct=notification-email&mt=8&pt=524675> or Android<https://play.google.com/store/apps/details?id=com.github.android&referrer=utm_campaign%3Dnotification-email%26utm_medium%3Demail%26utm_source%3Dgithub>.

**Comment by alastaira** [comment]
@edsonee LaserGRBL already shows the calculated percentage values of the supplied values as shown in the screenshot above, but you want the option to do it the other way around? (i.e. users type a percentage, and it converts to a value?). I suppose I could see a possible value in that, but I wonder if it would make the interface confusing to allow both methods.

**Comment by edsonee** [comment]
Alastaira you say: you want the option to do it the other way around? (i.e. users type a percentage, and it converts to a value?)
YES, that users enter percentage and next to this the value (proportional, see image)
Thanks Alastaria!
(My native language is not English, maybe I don't express ideas well).

If you think of it as a programmer this modification is unnecessary but if you think of it as someone who is going to use LASERGRBL for the first time, it will help him.

**Comment by edsonee** [comment]
MENU SVG
![2](https://user-images.githubusercontent.com/83145998/135733131-a9e3c10b-c454-4cec-b3b4-66a19e41e30c.JPG)

**Comment by arkypita** [comment]
> MENU SVG ![2](https://user-images.githubusercontent.com/83145998/135733131-a9e3c10b-c454-4cec-b3b4-66a19e41e30c.JPG)

Hi

For consistency I should add percentage also in SVG menu.

M3 is necessary (you can select M3/M4 for SVG too). M5 it is always superfluous since there is no other option, but always for the same reason does not bother.

Regarding the request to modify the UI to work directly in percentage I reply that it is something that I had assumed but that I have discarded because, working in percentage, requires the assumption that the GRBL configuration values ($30, $31) are always known. 

This is not always true!

For example until you connect a machine for the first time, for example when you use LaserGRBL to generate GCODE for a machine connected to another system (for example because it is not a GRBL machine), for example for those firmware (marlin, smoothie) for which there is experimental support (perhaps not very functional) but which do not plan to have the parameters $ 30 and $ 31.

For all these situations the choice to use the S-Value is much more correct and opens up more possibilities to use a percentage value.

**Reply by arkypita** [reply]
UI consistency implemented in v4.6.0 currently pre-release, can be downloaded here for test: https://github.com/arkypita/LaserGRBL/releases/download/v4.6.0/install.exe
